### PR TITLE
drivers: pinmux: stm32: st: add L4 UART4 AF on PC10 and PC11

### PR DESCRIPTION
**Pull request by GeorgeCGV**
- Added UART4 alternate pin function for L4 µC for PC10 and PC11.
- Corrected naming of previously defined UART4 TX and RX defines.
  Was defined as U**S**ART but in reallity it is only UART.

Due to naming correction had to touch `pinmux.c` of `disco_l475_iot1` board.

### CODE REVIEW

**Comment by GeorgeCGV** [comment]
@erwango

File: drivers/pinmux/stm32/pinmux_stm32l4x.h [hunk 1]
```diff
@@ -108,10 +108,14 @@
 	(STM32_PINMUX_ALT_FUNC_6 | STM32_PUSHPULL_NOPULL)
 #define STM32L4X_PINMUX_FUNC_PC10_USART3_TX \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
+#define STM32L4X_PINMUX_FUNC_PC10_UART4_TX \
+	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
```
Comment (erwango) [inline drivers/pinmux/stm32/pinmux_stm32l4x.h]: STM32_PINMUX_ALT_FUNC_8

File: drivers/pinmux/stm32/pinmux_stm32l4x.h [hunk 2]
```diff
@@ -108,10 +108,14 @@
...
 #define STM32L4X_PINMUX_FUNC_PC11_SPI3_MISO \
 	(STM32_PINMUX_ALT_FUNC_6 | STM32_PUSHPULL_NOPULL)
 #define STM32L4X_PINMUX_FUNC_PC11_USART3_RX \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUPDR_NO_PULL)
+#define STM32L4X_PINMUX_FUNC_PC11_UART4_RX \
+	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUPDR_NO_PULL)
```
Comment (erwango) [inline drivers/pinmux/stm32/pinmux_stm32l4x.h]: STM32_PINMUX_ALT_FUNC_8

File: dts/arm/st/l4/stm32l4-pinctrl.dtsi [hunk 3]
```diff
@@ -70,6 +70,12 @@
 					tx = <STM32_PIN_PA0 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_NOPULL)>;
 				};
 			};
+			uart4_pins_b: uart4_b {
+				rx_tx {
+					rx = <STM32_PIN_PC11 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUPDR_NO_PULL)>;
+					tx = <STM32_PIN_PC10 (STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_NOPULL)>;
```
Comment (erwango) [inline dts/arm/st/l4/stm32l4-pinctrl.dtsi]: STM32_PINMUX_ALT_FUNC_8 for both pins

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by GeorgeCGV** [comment]
@erwango you are right it shall be AF8, corrected

**Review by erwango — APPROVED** [review]

**Pull request merged**